unit-tests/log/test-multi-thread-logs.cpp

```diff
@@ -0,0 +1,231 @@
+int get_log_ideal_time(int first_time_ms, int first_value, int second_time_ms, int second_value, int value_in_log)
+{
+    int ideal_time = value_in_log * (second_time_ms - first_time_ms) / (second_value - first_value) +
+        (first_time_ms * (second_value / (second_value - first_value))) - second_time_ms * (first_value / (second_value - first_value));
+    
+    return ideal_time;
+}
+
+std::pair<unsigned int, int> get_next_50_lines_time_val(std::ifstream& ifs)
+{
+    unsigned int avg_time_ms = -1;
+    int avg_value = -1;
+    for (int i = 0; i < 50; ++i) {
+        std::string line;
+        getline(ifs, line);
+        avg_time_ms += get_time_from_line(line);
+        avg_value += stoi(line.substr(line.find_last_of(" ")));
+    }
+    avg_time_ms /= 50;
+    avg_value /= 50;
+    return std::make_pair(avg_time_ms, avg_value);
+}
+
+//----------------------- HELPER METHODS END ---------------------------
+
+TEST_CASE("async logger", "[log][async_log]")
+{
+    size_t n_callbacks = 0;
+    auto callback = [&](rs2_log_severity severity, rs2::log_message const& msg)
+    {
+        ++n_callbacks;
+        std::cout << msg.raw();
+    };
+    
+    auto func = [](int required_value) {
+        int iterations = 0;
+        auto start = std::chrono::steady_clock::now();
+        
+        while (iterations < number_of_iterations)
+        {
+            std::stringstream ss;
+            ss << "atomic integer = " << ++atomic_integer;
+            auto start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            rs2::log(RS2_LOG_SEVERITY_DEBUG, ss.str().c_str());
+            std::chrono::milliseconds ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            std::chrono::milliseconds delta_ms = ms - start_ms;
+            if (delta_ms > max_time) max_time = delta_ms;
+            if (delta_ms < min_time) min_time = delta_ms;
+            avg_time += delta_ms;
+            start_ms = ms;
+            ++iterations;
+        }
+    };
+    const char* log_file_path = ".//multi-thread-logs.log";
+    try {
+        remove(log_file_path);
+    }
+    catch (...) {}
+    rs2::log_to_file(RS2_LOG_SEVERITY_DEBUG, log_file_path);
+
+    std::vector<std::thread> threads;
+
+    for (int i = 0; i < number_of_threads; ++i)
+    {
+        threads.push_back(std::thread(func, i + 1));
+    }
+    
+    for (auto&&t : threads)
+    {
+        if (t.joinable()) t.join();
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
+    std::stringstream ss;
+    ss << "max time = " << max_time.count() << " ms" << std::endl;
+    ss << "min time = " << min_time.count() << " ms" << std::endl;
+    ss << "avg time = " << (float) (avg_time.count()) / (number_of_iterations * number_of_threads) << " ms" << std::endl;
+    CAPTURE(ss.str());
+    CHECK_NOFAIL(max_time.count() < checked_log_write_time);
```

Comment:
Why both CHECK_NOFAIL and REQUIRE?

Reply:
The CHECK is with the threshold we would like to pass - it passes locally on my linux PC.
The REQUIRE is with the threshold that passes in the CI.

Reply:
Ummm... what? Different thresholds??

Reply:
yes - let's talk about it.